src/util/codec/datum.rs

```diff
@@ -153,7 +166,7 @@ impl Datum {
 
     fn cmp_f64(&self, ctx: &EvalContext, f: f64) -> Result<Ordering> {
         match *self {
-            Datum::Null | Datum::Min => Ok(Ordering::Less),
+            Datum::Null | Datum::Min | Datum::Json(_) => Ok(Ordering::Less),
```

Comment:
It seems `self` wouldn't be `Datum::Json` here.  And for this case in `tidb` it returns less(https://github.com/pingcap/tidb/blob/master/util/types/datum.go#L496), I'm not sure whether it's ok to convert `f` to `Datum::Json`  @hicqu @hhkbp2